### PR TITLE
Issue #4414: "Channel type not available" summarization error (release/0.31)

### PR DESCRIPTION
**Pull request by vladsud**

### CODE REVIEW

File: packages/runtime/datastore/src/dataStoreRuntime.ts [hunk 1]
```diff
@@ -202,6 +202,9 @@ IFluidDataStoreChannel, IFluidDataStoreRuntime, IFluidHandleContext {
         // Must always receive the data store type inside of the attributes
         if (tree?.trees !== undefined) {
             Object.keys(tree.trees).forEach((path) => {
+                // Issue #4414
+                if (path === "_search") { return; }
```
Comment (agarwal-navin) [inline packages/runtime/datastore/src/dataStoreRuntime.ts:206]: Where does this search blob come from? Does server write it?

File: packages/runtime/datastore/src/dataStoreRuntime.ts [hunk 2]
```diff
@@ -202,6 +202,9 @@ IFluidDataStoreChannel, IFluidDataStoreRuntime, IFluidHandleContext {
         // Must always receive the data store type inside of the attributes
         if (tree?.trees !== undefined) {
             Object.keys(tree.trees).forEach((path) => {
+                // Issue #4414
```
Comment (agarwal-navin) [inline packages/runtime/datastore/src/dataStoreRuntime.ts:205]: Can you please add a comment why we need to skip search blobs. Also, I think a link to the issue would be better as it makes it easier to navigate.

File: packages/runtime/datastore/src/dataStoreRuntime.ts (hunk 1 above)
Comment (vladsud) [reply packages/runtime/datastore/src/dataStoreRuntime.ts:206]: Bohemia writes it. That's the reason this code ideally should not be here. I.e. this is temp solution, I'll keep an issue opened and will remove it after Arin finishes his work to separate snapshot paths for user data (like _search) vs. channels (DDS)

Comment (agarwal-navin) [reply packages/runtime/datastore/src/dataStoreRuntime.ts:206]: Yeah, that makes sense. Can you please add a link to the issue here.

**Review by agarwal-navin — APPROVED** [review]

**Pull request merged**